runtime-modules/content/src/lib.rs

```diff
@@ -0,0 +1,1142 @@
+    BalanceOf<T>,
+>;
+
+/// Information about channel being created.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct ChannelCreationParameters<ChannelCategoryId> {
+    /// ChannelCategory to enter the channel into.
+    in_category: ChannelCategoryId,
+    /// Metadata about the channel.
+    meta: Vec<u8>,
+}
+
+/// Information about channel being updated.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct ChannelUpdateParameters<ChannelCategoryId> {
+    /// If set, the new channel category to move the channel into.
+    new_in_category: Option<ChannelCategoryId>,
+    /// If set, metadata update for the channel.
+    new_meta: Option<Vec<u8>>,
+}
+
+/// A category that videos can belong to. A category will not be deleted if it contains any videos.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct VideoCategory {
+    /// The number of videos in this category.
+    number_of_videos_in_category: u32,
+}
+
+/// Information about the video category being created.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct VideoCategoryCreationParameters {
+    /// Metadata about the video category.
+    meta: Vec<u8>,
+}
+
+/// Information about the video category being updated.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct VideoCategoryUpdateParameters {
+    // Because it is the only field it is not an Option
+    /// Metadata update for the video category.
+    new_meta: Vec<u8>,
+}
+
+/// Information about the video being created.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct VideoCreationParameters<VideoCategoryId> {
+    /// The video category the video belongs to.
+    in_category: VideoCategoryId,
+    /// Metadata for the video.
+    meta: Vec<u8>,
+}
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct VideoUpdateParameters<VideoCategoryId> {
+    /// If set, the new category the video should be moved into.
+    new_in_category: Option<VideoCategoryId>,
+    /// If set, metadata update for the video.
+    new_meta: Option<Vec<u8>>,
+}
+
+/// A video which belongs to a channel. A video may be part of a series or playlist.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct Video<ChannelId, SeriesId, PlaylistId> {
+    in_channel: ChannelId,
+    // keep track of which seasons and playlists which reference the video
```

Comment:
Okay that makes sense. I was thinking of playlists really as a very simple series of videos. A series with a single season basically.  (interestingly a series can be though of as a playlist of playlists and perhaps even a series could be fully described in metadata and doesn't really need a lot of runtime state, but that would mean series would also not be constrained to only being composed of channel videos)

So the playlist you are describing is really a "curated list" of videos. In such a case does it make sense to allow a playlist to live "outside" of a channel?

Reply:
It could in theory, but Atlas is not designed for that, so would rather not try to go down that path now.

But you will need to update the metadata then for this?

Reply:
Yes now that you merged the protobuf PR I can update the metadata message in this PR.

Reply:
done in e19e1f677e2f3fd588ff7dd3d2446ca8b5538fc0

Reply:
done in 41af8a4ff0a47e9171763841be202aac4e4f9067 and 2a03591a491e1e1de3e1e898f192bc3dadfedced